### PR TITLE
Setup `react-router-dom` and document contribution guidelines

### PR DESCRIPTION
**Pull request by codeAesthetic**
to be merged after #2

### CODE REVIEW

**Comment by chirgjn** [comment]
Please request review only after dependent PRs are checked in

File: README.md [hunk 1]
```diff
@@ -7,3 +7,88 @@ command used for creating typescript template `create-react-app my-app --templat
...
+
+  - If an interface is specific only to the component, and cannot be used anywhere else
+    - If an interface is small and a component needs 1-2 such interfaces, then the interface can be defined inside the component
+    - Otherwise, use `<component_name>.interfaces.ts` file in the same folder as the component
+  - If an interface can be used by multiple components, methods etc (eg: module specific interface, api related interfaces), define them inside `src/common/interface.ts` folder
+
+- Always use hooks. Do not create any component in classes
+
+  - Older components can remain class based, but no new component should be in class
+  - Never break the rules of hooks - https://reactjs.org/docs/hooks-rules.html.
```
Comment (chirgjn) [inline README.md:37]: I think this is redundant because we assume that the contributor has some React experience.
Moreover any react codebase will need to adhere to these rules so it's a bit implicit.

File: src/common/routes.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import React from "react";
+
+import IRoute from "./interface";
+import * as pageRoutes from "./pageRoutes";
+//Lazy routes
+const HomePage = React.lazy(() => import("../Home/Home"));
+const AboutPage = React.lazy(() => import("../About/About"));
+const Auth = React.lazy(() => import("../Auth/Auth"));
```
Comment (chirgjn) [inline src/common/routes.ts:8]: I wasn't able to find `src/Auth` in the codebase

File: src/common/routes.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import React from "react";
+
+import IRoute from "./interface";
+import * as pageRoutes from "./pageRoutes";
+//Lazy routes
+const HomePage = React.lazy(() => import("../Home/Home"));
```
Comment (chirgjn) [inline src/common/routes.ts:6]: I prefer absolute paths over relative ones. It makes it easier to move files/modules around in a predictable fashion.

Can you change these and add the same to the documentation.

**Review by chirgjn — CHANGES_REQUESTED** [review]
- It should be `src/Application.tsx` not `src/application.tsx`
- Use Headers instead of nested list, whenever possible
- I think a lot of this documentation can be put inside: `CONTRIBUTING.md`
- Remove `.DS_Store`

File: README.md [hunk 4]
```diff
@@ -7,3 +7,88 @@ command used for creating typescript template `create-react-app my-app --templat
...
+  - common folder will contains utilities that is common across multiple components.
+  - Component specific utilities will reside under their folder itself.
+
+```
+  ├── src
+  │    ├──Auth (Directory name)
+       |   ├── Login.tsx
+       |   ├── Logout.tsx
+       |   ├── Auth.interface.ts (interface and types for components)
+       |   └── Auth.style.css (Style)
```
Comment (chirgjn) [inline README.md:74]: Let's add `*.style.css` files to the structure in #4

File: README.md [hunk 5]
```diff
@@ -7,3 +7,88 @@ command used for creating typescript template `create-react-app my-app --templat
...
+
+  - If a utility method is specific only to the component, has logic specific to the component and cannot be used anywhere else, add it to `<component_name>.utils.ts` in the same folder as the component.
+  - Otherwise, add it to one of the files inside `src/common/utils.ts` folder.
+
+- Interfaces file
+
+  - If an interface is specific only to the component, and cannot be used anywhere else
+    - If an interface is small and a component needs 1-2 such interfaces, then the interface can be defined inside the component
+    - Otherwise, use `<component_name>.interfaces.ts` file in the same folder as the component
+  - If an interface can be used by multiple components, methods etc (eg: module specific interface, api related interfaces), define them inside `src/common/interface.ts` folder
```
[the remark this answers is not in the document]
Comment (chirgjn) [reply README.md]: This hasn't been addressed

File: README.md (hunk 1 above)
Comment (chirgjn) [reply README.md]: This hasn't been addressed

File: README.md [hunk 6]
```diff
@@ -4,5 +4,5 @@ The project uses [Create React App](https://create-react-app.dev/)
 
 ## How to run the project
 
-- Run `yarn install` to install all required project dependencies
-- Run `yarn start` to run project
+- run `yarn install` to install all required project dependencies.
+- run `yarn start` to run project.
```
Comment (chirgjn) [inline README.md:8]: This seems unrelated

File: src/common/routes.ts (hunk 2 above)
Comment (chirgjn) [reply src/common/routes.ts]: This hasn't been addressed

File: README.md (hunk 4 above)
Comment (chirgjn) [reply README.md]: This hasn't been addressed

**Review by chirgjn — CHANGES_REQUESTED** [review]
> - It should be src/Application.tsx not src/application.tsx
> - Use Headers instead of nested list, whenever possible

File: README.md (hunk 6 above)
Comment (codeAesthetic) [reply README.md:8]: Do you want me to add a PR just for removing this?

**Comment by codeAesthetic** [comment]
> > * It should be src/Application.tsx not src/application.tsx
> > * Use Headers instead of nested list, whenever possible

didn't get this point.

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (codeAesthetic) [reply README.md]: Are you looking at some previous commit because this has been changed.

File: src/common/routes.ts (hunk 2 above)
Comment (codeAesthetic) [reply src/common/routes.ts]: this will be added later

**Comment by codeAesthetic** [comment]
I have renamed file `application.tsx` to `Application.tsx` but its not reflecting in the repository in vscode its showing `Application`.Is this some kind of issue at my end or at github's end?
Screenshot 2021-06-06 at 2 29 08 AM

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,89 +1,90 @@
-# Coding guidelines
+# Contirbution guidelines
```
Comment (chirgjn) [inline CONTRIBUTING.md:1]: This is what I was expecting. Can you take a look at the changes and resolve this comments once you've done so.

**Review by chirgjn — CHANGES_REQUESTED** [review]
